src/components/Resource/ResourceCommentSection.tsx

```diff
@@ -1,84 +1,123 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
 import { t } from "i18next";
+import { useQueryParams } from "raviger";
 import { useState } from "react";
 import { toast } from "sonner";
 
-import PaginatedList from "@/CAREUI/misc/PaginatedList";
+import { cn } from "@/lib/utils";
 
 import { Button } from "@/components/ui/button";
 import { Textarea } from "@/components/ui/textarea";
 
 import { Avatar } from "@/components/Common/Avatar";
-import CircularProgress from "@/components/Common/CircularProgress";
+import PaginationComponent from "@/components/Common/Pagination";
+import { CardListSkeleton } from "@/components/Common/SkeletonLoading";
 
 import routes from "@/Utils/request/api";
-import request from "@/Utils/request/request";
+import mutate from "@/Utils/request/mutate";
+import query from "@/Utils/request/query";
 import { formatName, relativeTime } from "@/Utils/utils";
 import { CommentModel } from "@/types/resourceRequest/resourceRequest";
 
 const CommentSection = (props: { id: string }) => {
+  const { id } = props;
   const [commentBox, setCommentBox] = useState("");
+  const queryClient = useQueryClient();
 
-  const onSubmitComment = async () => {
-    const payload = {
-      comment: commentBox,
-    };
+  const [qParams, setQueryParams] = useQueryParams();
+
+  const { data: resourceComments, isLoading } = useQuery({
+    queryKey: ["resourceComments", id, qParams],
+    queryFn: query(routes.getResourceComments, {
+      queryParams: {
+        limit: 15,
+        offset: ((qParams.page ?? 1) - 1) * 15,
+      },
+      pathParams: { id },
+    }),
+  });
+
+  const { mutate: addComment } = useMutation({
+    mutationFn: mutate(routes.addResourceComments, {
+      pathParams: { id },
+    }),
+    onSuccess: () => {
+      toast.success(t("comment_added_successfully"));
+      queryClient.invalidateQueries({ queryKey: ["resourceComments"] });
```

Comment:
we don't need to invalidate all resource comments right? only the one associated to the particular resource right?

```suggestion
      queryClient.invalidateQueries({ queryKey: ["resourceComments", id] });
```